jobs/src/main/scala/dcos/metronome/jobspec/impl/JobSpecSchedulerActor.scala

```diff
@@ -49,10 +50,27 @@ class JobSpecSchedulerActor(
 
   def runJob(schedule: ScheduleSpec): Unit = {
     log.info(s"Start next run of job ${spec.id}, which was scheduled for $scheduledAt")
-    runService.startJobRun(spec, Some(schedule.startingDeadline))
+
+    if (!(schedule.concurrencyPolicy == ConcurrencyPolicy.Forbid && activeRunsForJobSpec())) {
+      runService.startJobRun(spec, Some(schedule.startingDeadline))
+    } else {
+      log.debug(s"Skipping Scheduled run for ${spec.id} based on concurrency policy")
+    }
     scheduleNextRun()
   }
 
+  def activeRunsForJobSpec(): Boolean = {
+    try {
+      val runResult = Await.result(runService.activeRuns(spec.id), 5.seconds)
```

Comment:
This is a bit dangerous to do blocking calls inside the receive method of the actor. How about piping the future back to the actor?